swift/Lottie.swift

```diff
@@ -0,0 +1,152 @@
+import CoreGraphics
+import CoreMedia
+
+import thorvg
+
+/// Errors that can occur while working with Lottie animations.
+public enum LottieError: Error {
+    case failedToLoadLottieFromString
+    case failedToLoadLottieFromPath
+    case failedToDrawLottieFrameOntoCanvas
+    case failedToInitialiseTvgEngine
+    case frameIndexOutOfBounds
+}
+
+/// Shorthand for the buffer type used to represent image pixel data.
+public typealias Buffer = UnsafeMutablePointer<UInt32>
+
+/// Shorthand for a pointer to an internal Lottie animation object
+private typealias Animation = OpaquePointer
+
+/// Shorthand for a pointer to an internal rendering canvas
+private typealias Canvas = OpaquePointer
+
+/// Object used to load and render Lottie frames.
+public class Lottie {
+    /// The number of frames in the Lottie animation.
+    public let numberOfFrames: Int
+
+    /// The duration of the Lottie animation.
+    public let duration: CMTime
+
+    /// The Lottie animation, used for manipulating and rendering frames.
+    private let animation: Animation?
+
+    /// The canvas used for rendering the animation frames, allowing reuse across renders.
+    private var canvas: Canvas? = nil
+
+    /// Initializes a new Lottie instance from a file path.
+    /// - Parameters:
+    ///   - path: The file path of the Lottie animation.
+    public convenience init(path: String) throws {
```

Comment:
I like the idea of `URL`, but I'm a bit nervous of handing over the responsibility of loading the URL contents to `Lottie`. With the `path` and `string` based initializers, it is really clear exactly what form you need the data in, and they map directly to the ThorVG initializers as well. 

For now, I might leave these as is, just to not have to add more complexity when it comes to validating we can load the URL data correctly. Because I would also imagine that if we have a URL we may have remote content that we'd need to fetch as well.